jetty-util/src/main/java/org/eclipse/jetty/util/resource/FileResource.java

```diff
@@ -65,6 +69,7 @@ public FileResource(URL url)
         {
             // Try standard API to convert URL to file.
             file =new File(url.toURI());
+            assertValidPath(file.toString());
```

Comment:
I'm not sure we should  be checking this path.   If it is a valid URI obtained from a valid URL, then it should be a valid resource.

Isn't it only the addPath path we need to be concerned about?